### PR TITLE
Replace Internal.AspNetCore.Sdk with Microsoft.DotNet.Arcade.Sdk

### PR DESCRIPTION
**Pull request by natemcmaster**
This is part of #7280. 

Here are the high-level changes:
* Change location of dotnet.exe to `$repoRoot/dotnet.exe` instead of `$repoRoot/x64/dotnet.exe`
* Remove our own implementation of versioning properties and targets
* Remove licenseUrl from all nuget packages and use PackageLicenseExpression instead
* Remove a bunch of properties and targets from our code that are provided by Microsoft.DotNet.Arcade.Sdk (or are unnecessary now)
* Add a handful of workaround to override defaults from Arcade which don't work for this repo

Work remaining to complete #7280 (will be done in a separate PR)
* Replace KoreBuild usages in build scripts with Arcade scripts
* Update Azure pipelines to use Arcade YAML templates

FYI - marking this as a draft PR until I can get CI to pass. Local builds are working, but still need to validate CI output and tests don't change substantially.

/cc @Pilchie @markwilkie @tmat @aspnet/build

### CODE REVIEW

**Review by Pilchie — APPROVED** [review]
Can't say I know KoreBuild well enough, or reviewed this closely enough to be super confident, but glad to see it happening!

**Comment by anurse** [comment]
Farewell (almost) KoreBuild, you served us well :).

**Review by analogrelay — APPROVED** [review]
Mostly in the same boat as @Pilchie, a lot of this is beyond what I was familiar with. But glad to see us get closer in alignment with Arcade :). Good to do this early in preview 7 too so there's some time to shake out the bugs.

**Review by analogrelay — APPROVED** [review]
Mostly in the same boat as @Pilchie, a lot of this is beyond what I was familiar with. But glad to see us get closer in alignment with Arcade :). Good to do this early in preview 7 too so there's some time to shake out the bugs.

**Review by ryanbrandenburg — APPROVED** [review]
I didn't spot any problems, but this is a big one with a lot of moving parts, so I feel like the proof will mostly be in if it works or not.

File: activate.ps1 [hunk 1]
```diff
@@ -37,7 +37,7 @@ deactivate -init
 
 $_OLD_PATH = $env:PATH
 # Tell dotnet where to find itself
-$env:DOTNET_ROOT = "$PSScriptRoot\.dotnet\x64"
+$env:DOTNET_ROOT = "$PSScriptRoot\.dotnet"
```
Comment (jkotalik) [inline activate.ps1]: Does this mean we don't have the x86 version of dotnet anymore?

File: build.ps1 [hunk 2]
```diff
@@ -348,6 +348,10 @@ if ($RunBuild -and ($All -or $BuildJava) -and -not $NoBuildJava) {
         }
     }
 
+    if ($env:PATH -notlike "*${env:JAVA_HOME}*") {
```
Comment (jkotalik) [inline build.ps1]: Clarify/comment on what this is for?

File: build/AzureIntegration.targets [hunk 3]
```diff
@@ -6,12 +6,8 @@
   <Target Name="BuildAzureIntegration">
     <PropertyGroup>
       <AzureIntegrationProjProperties>
-        AspNetUniverseBuildOffline=true;
         RepoRoot=$(AzureIntegrationProjectRoot);
-        VersionSuffix=$(VersionSuffix);
-        BuildNumberSuffix=$(BuildNumberSuffix);
```
Comment (jkotalik) [inline build/AzureIntegration.targets]: Are these properties still available to the Azure integration projects?

Comment (natemcmaster) [reply build/AzureIntegration.targets]: VersionSuffix should be - that's a built in SDK property. BuildNumberSuffix is being removed, but I didn't see it getting used anywhere in Azure integration.

File: activate.ps1 (hunk 1 above)
Comment (natemcmaster) [reply activate.ps1]: We do, but its in .dotnet\x86\.

Which reminds me, I'm going to update this PR to also set the `DOTNET_ROOT(x86)` env var from these scripts

File: build.ps1 (hunk 2 above)
Comment (natemcmaster) [reply build.ps1]: It's a minor cleanup that fixes https://github.com/aspnet/AspNetCore-Internal/issues/2592.

**Comment by natemcmaster** [comment]
> I didn't spot any problems, but this is a big one with a lot of moving parts, so I feel like the proof will mostly be in if it works or not.

Agreed. I'm doing a thorough diff of the build outputs to make sure things like nuspecs, manifests, filenames, etc. didn't change unexpectedly. 

@ryanbrandenburg - I'm getting consistent test failures on non-Windows platforms in the templates. I couldn't figure out how my changes would cause this failure. Are test failures here a known issue?
https://dev.azure.com/dnceng/public/_build/results?buildId=214984&view=ms.vss-test-web.build-test-results-tab&runId=5269634&resultId=110652&paneView=debug
image

**Comment by natemcmaster** [comment]
Blocked on https://github.com/dotnet/arcade/issues/2984 - haven't found a workaround. Struggling to figure out to make our .wixproj files work without some nasty hacks. Evaluation order for VersionSuffix moved from Directory.Build.props to Microsoft.Common.targets, which makes it hard to use `$(PackageVersion)` in the output file name for a Windows installer.

**Comment by ryanbrandenburg** [comment]
@natemcmaster The templating link failures should be fixed by https://github.com/aspnet/AspNetCore/pull/10957 if you rebase on to latest.

**Comment by natemcmaster** [comment]
Thanks Ryan! I'll update and try again

**Comment by natemcmaster** [comment]
I think I found a workaround for https://github.com/dotnet/arcade/issues/2984 and have pushed a fix to the .wixproj files.

**Comment by natemcmaster** [comment]
@aspnet-hello triage https://dev.azure.com/dnceng/public/_build/results?buildId=216603

**Comment by aspnet-hello** [comment]
This comment was made automatically. If there is a problem contact aspnetcore-build@microsoft.com.

I've triaged the above build. I've created/commented on the following issue(s)
https://github.com/aspnet/AspNetCore-Internal/issues/2646

**Comment by natemcmaster** [comment]
This is a big PR. I've carefully looked over the build outputs and don't see any major unexpected changes. Merging now so we can move forward with the next round of Arcade improvements.

**Pull request merged**